### PR TITLE
ignore getReplica's error when handle rebalanceTask

### PR DESCRIPTION
**Pull request by jiaoew1991**
issue: #17367 
Signed-off-by: Enwei Jiao <jiaoew2011@gmail.com>

### CODE REVIEW

File: internal/querycoord/task.go [hunk 1]
```diff
@@ -1935,8 +1935,8 @@ func (lbt *loadBalanceTask) processNodeDownLoadBalance(ctx context.Context) erro
 			log.Info("loadBalanceTask: get collection's all partitionIDs", zap.Int64("collectionID", collectionID), zap.Int64s("partitionIDs", toRecoverPartitionIDs))
 			replica, err := lbt.getReplica(nodeID, collectionID)
 			if err != nil {
-				lbt.setResultInfo(err)
-				return err
+				// getReplica maybe failed, it will cause the balanceTask execute infinitely
+				log.Warn("loadBalanceTask: get replica failed", zap.Int64("collectionID", collectionID), zap.Int64("nodeId", nodeID))
```
Comment (congqixia) [inline internal/querycoord/task.go:1939]: Why `getReplica` failed at the first place?

Comment (congqixia) [reply internal/querycoord/task.go:1939]: If `getReplica` returns error, it's not safe to use `replica` any more(it shall be nil).

Comment (yah01) [reply internal/querycoord/task.go:1939]: > Why `getReplica` failed at the first place?

Inconsistency happens if the QueryCoord crashes while the LoadBalance task has modified the replica, but itself is not removed from etcd. The recovered task will never get replica with the nodeID and collectionID, and retry infinitely.

**Review by congqixia — APPROVED** [review]
/lgtm
NIT: maybe we need more comment here to explain why this error could be ignored.

**Pull request merged**